### PR TITLE
Draw border on the left side of quote in Reader

### LINKED ISSUE #9881: Reader: Improve how quotes are displayed in posts
### Expected behavior
Quotes on a Gutenberg post should display correctly on Reader.

### Actual behavior
They don't look like quotes at all.

img_1255

### Steps to reproduce the behavior
Check out a post with quotes on reader.

### PR DESCRIPTION
**Pull request by jklausa**
Fixes #9881

This is a draft PR because I think I've hit a wall with my current TextKit knowledge and need another pair of eyes on this to avoid going bonkers (well, more bonkers than I currently am).

Here's what this PR accomplishes — note the beautiful nice border on the left side of the quote.

 
However, there are some issues with image drawing. Namely, the existing code path to find out the size for the image doesn't work. I figured a workaround, but it doesn't produce ideal results — note the image doesn't stretch all the way to the right of the view — and it's even more broken in the landscape.

I appreciate any and all pointers on how to continue with this — tagging both you @diegoreymendez, since you are the person that probably knows the most about TextKit 'round these parts, but also @aerych since I think you dabbled with Reader parts of it too.

Screenshots showing the issues in current state:
Simulator Screen Shot - iPhone Xs Max - 2019-04-04 at 09 35 44
Simulator Screen Shot - iPhone Xs Max - 2019-04-04 at 09 35 36

To test:

Update release notes:

- [ ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

**Comment by aerych** [comment]
Hey @jklausa!  
I poked at this branch for a while in the simulator and then did some comparisons between the reader with and with out the patch.  Everything was looking good to me and I wasn't able to reproduce the image sizing issue shown in the screen shots above.  
Is it possible there was something left over from an other edit causing a conflict?   Was there a certain device that the glitch was happening on?

**Comment by jklausa** [comment]
@aerych huh! that's surprising to me. I was testing mostly on XS Max, so maybe it is device specific. I'll check again. I _think_ I blew away all Derived Data etc to make sure I didn't accidentally stumble into a valid implementation, but I'll quadruple check again.

**Comment by jklausa** [comment]
I triple-quadruple checked and this is definitely still happening in this PR, however on smaller screen it's not as pronounced as on Xs Max or iPads.

I left some screenshots to illustrate the issue on iPads. I'm out of ideas — @diegoreymendez, could I bother you to take a peek at this?

| Changed | Original |
|--------|-------|
|Simulator Screen Shot - iPad Pro (11-inch) - 2019-04-05 at 22 51 23 |  Simulator Screen Shot - iPad Pro (11-inch) - 2019-04-05 at 22 52 58 |
|Simulator Screen Shot - iPad Pro (11-inch) - 2019-04-05 at 22 51 22 |  Simulator Screen Shot - iPad Pro (11-inch) - 2019-04-05 at 22 53 01 |

**Comment by aerych** [comment]
I guess I wasn't seeing the issue because I was testing with either the iPhone 8 or iPhone X simulator. 

I checked out a fresh copy of the branch, deleted derived data, and tried again in the iPad Pro 12.9 inch simulator.  I'm stuck getting the following exception:

Screen Shot 2019-04-05 at 4 25 24 PM

The stack trace isn't terribly helpful as to what the issue is.  Maybe Xcode 10.1 is a factor?  I've had some weird Simulator glitches recently so I'm not convinced that's not the cause.

I switched to the iPhone XS Max simulator which is not crashing, and noticed this interesting bit of behavior.   Steps to reproduce:
- From Portrait orientation tap on a post to view the reader detail. 
- Rotate to landscape.  Notice the image is not full width. 
- Rotate back to portrait.
- Rotate back to landscape. Interestingly, the detail is no longer full screen and we see the split view. 
- Notice that the image is the correct width for the split view. 
- Notice that the image appears like it was always being sized for split view when rotating when comparing the screen shots side by side. 
Simulator Screen Shot - iPhone XS Max - 2019-04-05 at 16 31 39
Simulator Screen Shot - iPhone XS Max - 2019-04-05 at 16 31 45

Interesting!  I double checked and I see this same behavior in the reader without this patch on the iPhone XS Max.  Its a glitch but not related to the patch. 

I switched to the iPad Pro 9.7 inch model. This still crashes for me 🤔.  I commented out the changes that assigned the custom LayoutManager in WPRichContentView and tried again. If it crashed it might have just been the simulator being weird.   No crash but when I repeated the above test, rotating from portrait to landscape and back and again the image behavior was the same but the master viewcontroller for the split view never appeared.  S
Simulator Screen Shot - iPad Pro (9 7 inch) - 2019-04-05 at 16 52 08
Simulator Screen Shot - iPad Pro (9 7 inch) - 2019-04-05 at 16 52 10
I tried it on a few different posts and it was the same behavior. 

I'm going to switch computers and do a little more debugging.   More to follow.

**Comment by aerych** [comment]
Switched computers. Ran `rake dependencies` to update all the things. Checked out the branch. Repeated test with the iPad Pro 12.9 inch simulator.   I got the same exception as shown above. 🤔   Not sure what to make of that.   This computer is also running Xcode 10.1 if it matters

**Comment by diegoreymendez** [comment]
I've been testing this for quite a while today, and I'm pretty convinced that the problems you're seeing are because replacing the layout manager after the text view is created is not very well supported (or at least the method we're using is not the right one).

I think one way to test this could be through a playground, setting the layout manager like we're doing here and then trying before the text view is created, and then comparing things a bit.

I'll pause for now though since it's late here.

**Comment by jklausa** [comment]
Hmm, that's a good hint! I'll try to poke around that

**Comment by diegoreymendez** [comment]
> Hmm, that's a good hint! I'll try to poke around that

Just to make it clear though - I didn't manage to resolve this in my tests.  I spent around 4 hours looking for alternatives.

I didn't try assigning the layout manager, storage and container when creating the text view as that would require changing things a bit more heavily.  I believe this is what may resolve the issue for us.

**Comment by jklausa** [comment]
So, thanks to @diegoreymendez I think I have this working now! It's still not shippable and needs cleaning up, but _I think_ it works 100% correctly now.

I could use some guidance in how to proceed here though.

If you look into the last commit, you'll see that I had to create instances of the `WPRichContentView` manually, instead of relying on storyboards. That's... not great, but maybe there's some way to work around it, that I'm not thinking of right now? 
The way I'm currently overriding the elements from `storyboards` obviously has got to. I don't want to manually layout the whole thing, so will have to come up with something clever there too.


I'm leaving this as-is now, since it's super late here, but I plan to clean this up and finish tomorrow.

**Comment by jklausa** [comment]
Alright, I think I have this cleaned up now and ready to go. @diegoreymendez  — would you mind taking another look at this?

**Comment by elibud** [comment]
This is great @jklausa, I can not tell you how irrationally annoyed I am at the lack of that vertical line.

File: WordPress/Classes/ViewRelated/Views/WPRichText/WPRichContentView.swift [hunk 1]
```diff
@@ -162,8 +182,10 @@ class WPRichContentView: UITextView {
     ///
     @objc func setupView() {
         // Because the attachment manager is a lazy property.
+        _ = customLayoutManager
```
[the remark this answers is not in the document]
Comment (jklausa) [reply WordPress/Classes/ViewRelated/Views/WPRichText/WPRichContentView.swift]: I think you were looking at a old checkout of this branch? It's not there in the latest commit.

File: WordPress/Classes/ViewRelated/Reader/ReaderCommentCell.swift [hunk 2]
```diff
@@ -68,9 +70,17 @@ class ReaderCommentCell: UITableViewCell {
     override func awakeFromNib() {
         super.awakeFromNib()
 
+        let newTextView = WPRichContentView(frame: .zero, textContainer: nil)
+        newTextView.isScrollEnabled = false
+        newTextView.translatesAutoresizingMaskIntoConstraints = false
+
+        parentStackView.insertArrangedSubview(newTextView, at: 1)
```
[the remark this answers is not in the document]
Comment (jklausa) [reply WordPress/Classes/ViewRelated/Reader/ReaderCommentCell.swift]: Good call!

**Comment by jklausa** [comment]
@diegoreymendez all done, ready for a final look :)

File: WordPress/Classes/ViewRelated/Views/WPRichText/WPRichContentView.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (diegoreymendez) [reply WordPress/Classes/ViewRelated/Views/WPRichText/WPRichContentView.swift]: Sorry, that was an old comment I never submitted / removed.

**Comment by diegoreymendez** [comment]
I'm taking over for now, as @jklausa is AFK for a while.

@aerych - I added some changes... can I ask you to check this PR?

**Comment by aerych** [comment]
@diegoreymendez Just took the branch for a test drive in the iPad Pro simulator.  Everything was working fine and blockquotes look good. The changes don't seem to have had any negative impact in comments or the post detail as far as I can see. :)  Nicely done. Kudos to you and @jklausa both! 
:shipit: when ready.

**Comment by diegoreymendez** [comment]
Honestly, kudos to @jklausa for the great initiative and push.

Also thanks a ton @aerych for the quick review!

**Comment by designsimply** [comment]
I tested this as part of 12.4 beta testing and it looks great! 😍

IMG_2742
<p style="clear:both;"><sub>Tested with WP Internal 12.4.0.20190505 on iPhone 6S iOS 12.2.</sub></p>

**Pull request merged**